prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -139,51 +141,72 @@ def get_output_mapping(self):
         logger.info(f"Using kubeconfig file: {kubeconfig_file}")
         try:
-            config.load_kube_config(
-                config_file=(
-                    os.path.abspath(kubeconfig_file)
-                    if kubeconfig_file != "~/.kube/config"
-                    else os.path.expanduser(kubeconfig_file)
-                ),
-                context=input_context,
-            )
-        except ConfigException:
-            # If the kubeconfig file is not found, try to use the in-cluster config
-            logger.info("Using in-cluster config")
-            config.load_incluster_config()
-            context = {
-                "name": "In-Cluster",
-                "context": {
-                    "cluster": "in-cluster",  # Placeholder, as the real cluster name is not available
-                    "user": "service-account-name",  # Also a placeholder
-                },
-            }
-        else:
+            if kubeconfig_content:
+                config.load_kube_config_from_dict(
+                    kubeconfig_content, context=input_context
+                )
+            else:
+                try:
+                    config.load_kube_config(
+                        config_file=(
+                            os.path.abspath(kubeconfig_file)
+                            if kubeconfig_file != "~/.kube/config"
+                            else os.path.expanduser(kubeconfig_file)
+                        ),
+                        context=input_context,
+                    )
+                except ConfigException:
+                    # If the kubeconfig file is not found, try to use the in-cluster config
+                    logger.info("Using in-cluster config")
+                    config.load_incluster_config()
+                    context = {
+                        "name": "In-Cluster",
+                        "context": {
+                            "cluster": "in-cluster",  # Placeholder, as the real cluster name is not available
+                            "user": "service-account-name",  # Also a placeholder
+                        },
+                    }
+                    return KubernetesSession(
+                        api_client=client.ApiClient(), context=context
+                    )
             if input_context:
                 contexts = config.list_kube_config_contexts()[0]
                 for context_item in contexts:
                     if context_item["name"] == input_context:
                         context = context_item
             else:
                 context = config.list_kube_config_contexts()[1]
-        return KubernetesSession(api_client=client.ApiClient(), context=context)
+            return KubernetesSession(api_client=client.ApiClient(), context=context)
+        except Exception as error:
+            logger.critical(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+            raise KubernetesError(
```

Comment:
```suggestion
            raise KubernetesSetUpSessionError(
```

Could you please use the exception created for the setup session?